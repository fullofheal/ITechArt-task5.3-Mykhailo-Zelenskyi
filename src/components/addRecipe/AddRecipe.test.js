import { cleanup, screen, render } from "@testing-library/react";
import '@testing-library/jest-dom';
import { BrowserRouter as Router } from "react-router-dom";
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../../store';
import addRecipeActions from '../../store/actions/addRecipeActions';
import AddRecipe from './AddRecipe';
import RecipesList from '../recipesList/RecipesList';

afterEach(() => {
  cleanup();
});

const customRecipe = {
  id: 777,
  country: 'customCountry',
  category: 'customCategory',
  instructions: 'custom instructions',
  name: 'customName',
  tags: 'customTag',
  ingredients: [['custom ingredient','custom measure']],
  picture: 'customUrl',
  video: 'customUrl'
};

describe('test if component is rendered on the page', () => {
  test('should render AddRecipe component', () => {
    render(
      <Provider store={store}>
        <AddRecipe />
      </Provider>
    );
    const addRecipeComponent = screen.getByTestId('AddRecipe');
    expect(addRecipeComponent).toBeInTheDocument();
  });
});


  
test('should add custom recipe and display in the list', () => {
  const recipe = {
    ...customRecipe
  };
  store.dispatch(addRecipeActions.addToLocal(recipe));
  render(
    <Provider store={store}>
      <Router>
        <RecipesList />
      </Router>
    </Provider>
  );
  expect(screen.getByText('customName')).toBeInTheDocument();
  expect(screen.getByText('Country of origin: customCountry')).toBeInTheDocument();
  expect(screen.getByText('Tags: customTag')).toBeInTheDocument();
});  

test('should properly update addRecipeReducer with fetched selected recipe details', () => {
  const fetchedRecipe = {
    idMeal: 777,
    strMeal: 'customName',
    strArea: 'customCountry',
    strCategory: 'customCategory',
    strInstructions: 'custom instructions',
    strIngredient1: 'custom ingredient',
    strMeasure1: 'custom measure',
    strMealThumb: 'customUrl',
    strYoutube: 'customUrl',
    strTags: 'customTag'
  };
  const transformedRecipe = {
    ...customRecipe
  };
  store.dispatch(addRecipeActions.addRecipeFetched(fetchedRecipe));
  const reducerDetails = store.getState().addRecipeReducer.details;
  expect(reducerDetails).toEqual(transformedRecipe);    
});

test('AddRecipe matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Router>
        <AddRecipe />
      </Router>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
