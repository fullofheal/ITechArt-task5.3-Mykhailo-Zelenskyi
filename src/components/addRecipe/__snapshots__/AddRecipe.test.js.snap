// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddRecipe matches snapshot 1`] = `
<div
  className="add__wrapper"
  data-testid="AddRecipe"
>
  <h3>
    Add your own recipe, or edit an uploaded one
  </h3>
  <form
    className="form"
    data-testid="EditRecipeForm"
    onSubmit={[Function]}
  >
    <div
      className="form__picture"
    >
      <img
        alt=""
        src=""
      />
      <label
        className="form__label"
        htmlFor="picture-url"
      >
        Picture url
      </label>
      <input
        name="picture-url"
        onChange={[Function]}
        placeholder="Provide valid image url"
        type="text"
        value=""
      />
    </div>
    <div>
      <label
        className="form__label"
        htmlFor="name"
      >
        Name
      </label>
      <input
        name="name"
        onChange={[Function]}
        pattern="[^#$^&*?><\\\\/0-9]+"
        placeholder="Recipe name"
        required={true}
        type="text"
        value=""
      />
      <span>
        Only letters
      </span>
    </div>
    <div>
      <label
        className="form__label"
        htmlFor="tags"
      >
        Tags
      </label>
      <input
        name="tags"
        onChange={[Function]}
        pattern="[A-Za-z, ]+"
        placeholder="List associated tags"
        type="text"
        value=""
      />
      <span>
        Only letters
      </span>
    </div>
    <div>
      <label
        className="form__label"
        htmlFor="country"
      >
        Country of origin
      </label>
      <input
        name="country"
        onChange={[Function]}
        pattern="[A-Za-z ]+"
        placeholder="List associated tags"
        type="text"
        value=""
      />
      <span>
        Only letters
      </span>
    </div>
    <label
      className="form__label"
      htmlFor="instructions"
    >
      Instructions
    </label>
    <textarea
      className="form__instructions"
      name="instructions"
      onChange={[Function]}
      placeholder="Provide detailed preparation instructions"
      required={true}
      value=""
    />
    <div
      className="add-form__ingredients"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Add new ingredient
      </button>
    </div>
    <label
      className="form__label-video"
      htmlFor="video-url"
    >
      Video url
    </label>
    <input
      name="video-url"
      onChange={[Function]}
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Submit recipe
    </button>
  </form>
</div>
`;
